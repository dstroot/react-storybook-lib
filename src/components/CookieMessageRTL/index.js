import React, { useState } from 'react';

const CookieMessageRTL = () => {
  const [visible, setVisible] = useState(true);

  const hide = () => {
    setVisible(false);
  };

  if (visible) {
    return (
      <div data-testid="cookie-message" className="fixed-bottom bg-dark">
        {/* <div className="container">
          <div className="row p-4">
            <div className="col-10 text-light">
              We use cookies to understand the performance of our web site,
              enable social media features, and serve more relevant content to
              you. We may also use cookies on our and our partners’ behalf to
              help us manage advertising and assess the performance of our
              campaigns. For further information please see our&nbsp;
              <a href="/cookiepolicy" title="Cookie Policy.">
                Cookie Policy
              </a>
              .
            </div>
            <div className="col-2">
              <button className="btn btn-light btn-sm" onClick={hide}>
                I Agree
              </button>
            </div>
          </div>
        </div> */}

        <table className="table table-borderless m-0">
          <tbody>
            <tr>
              <td className="text-light">
                We use cookies to understand the performance of our web site,
                enable social media features, and serve more relevant content to
                you. We may also use cookies on our and our partners’ behalf to
                help us manage advertising and assess the performance of our
                campaigns. For further information please see our&nbsp;
                <a href="/cookiepolicy" title="Cookie Policy.">
                  Cookie Policy
                </a>
                .
              </td>
              <td className="align-middle">
                <button className="btn btn-light btn-sm" onClick={hide}>
                  I Agree
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    );
  } else {
    return null;
  }
};

export default CookieMessageRTL;
