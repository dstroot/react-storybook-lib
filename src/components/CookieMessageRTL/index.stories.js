import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import CookieMessageRTL from '../CookieMessageRTL';

// CookieMessageRTL section
storiesOf('Pac Life|CookieMessageRTL', module)
  .addParameters({
    // default for all stories in this book
    info: {
      inline: true,
      header: false,
      text: `
        Example component to show a cookie message.
    `,
    },
  })
  .add('show CookieMessageRTL', () => <CookieMessageRTL />);
