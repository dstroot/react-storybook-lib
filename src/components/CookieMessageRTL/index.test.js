import React from 'react';

/**
 * The react-testing-library is a very lightweight solution for testing React components.
 * It provides light utility functions on top of react-dom and react-dom/test-utils,
 * in a way that encourages better testing practices.
 */

// React test library
import { render, fireEvent, cleanup } from 'react-testing-library';
// Custom jest matchers from jest-dom
import 'jest-dom/extend-expect';

// component to test
import CookieMessageRTL from '../CookieMessageRTL';

// automatically unmount and cleanup DOM after each test is finished.
afterEach(cleanup);

// tests
describe('CookieMessageRTL', () => {
  it('it should render', () => {
    /**
     * By default, react-testing-library will create a div and
     * append that div to the document.body and this is where
     * your react component will be rendered.
     */
    const { container } = render(<CookieMessageRTL />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('it should be at the bottom of the page', () => {
    const { getByTestId } = render(<CookieMessageRTL />);
    expect(getByTestId('cookie-message')).toHaveClass('fixed-bottom bg-dark');
  });

  it('it should contain the expected text', () => {
    const { getByText } = render(<CookieMessageRTL />);
    const element = getByText(
      `We use cookies to understand the performance of our web site, enable social media features, and serve more relevant content to you. We may also use cookies on our and our partnersâ€™ behalf to help us manage advertising and assess the performance of our campaigns. For further information please see our`,
      { exact: false }
    );
    expect(element).toBeInTheDocument();
  });

  it('it should have a link to /cookiepolicy', () => {
    const { getByTitle } = render(<CookieMessageRTL />);
    expect(getByTitle('Cookie Policy.')).toBeInTheDocument();
  });

  it('it should have a button that says "I agree"', () => {
    const { getByText } = render(<CookieMessageRTL />);
    const button = getByText('I Agree');
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('btn');
  });

  it('it should disappear when the button is clicked', () => {
    // setup
    const { getByText, container } = render(<CookieMessageRTL />);

    // assert - componment exists
    expect(container.firstChild).toBeInTheDocument();

    // action - click button
    const button = getByText('I Agree');
    fireEvent.click(button);

    // assert - component no longer exists in DOM
    expect(container.firstChild).toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });
});

/**
 * Custom matchers:
 *   - toBeDisabled
 *   - toBeEnabled
 *   - toBeEmpty
 *   - toBeInTheDocument
 *   - toBeInvalid
 *   - toBeRequired
 *   - toBeValid
 *   - toBeVisible
 *   - toContainElement
 *   - toContainHTML
 *   - toHaveAttribute
 *   - toHaveClass
 *   - toHaveFocus
 *   - toHaveFormValues
 *   - toHaveStyle
 *   - toHaveTextContent
 */

/**
  * Based on the Guiding Principles, your test should resemble how users 
  * interact with your code (component, page, etc.) as much as possible. 
  * With this in mind, we recommend this order of priority:

  * Queries Accessible to Everyone queries that reflect the experience 
  * of visual/mouse users as well as those that use assistive technology
  * 
  * **getByLabelText**: Only really good for form fields, but this is the 
  * number one method a user finds those elements, so it should be your top preference.
  * 
  * **getByPlaceholderText**: A placeholder is not a substitute for a label. But if 
  * that's all you have, then it's better than alternatives.
  * 
  * getByText: Not useful for forms, but this is the number 1 method a user 
  * finds other elements (like buttons to click), so it should be your top 
  * preference for non-form elements.
  * 
  * getByDisplayValue: The current value of a form element can be useful when 
  * navigating a page with filled-in values.
  * 
  * Semantic Queries HTML5 and ARIA compliant selectors. Note that the user 
  * experience of interacting with these attributes varies greatly across 
  * browsers and assistive technology.
  * 
  * **getByAltText**: If your element is one which supports alt text (img, area, 
  * and input), then you can use this to find that element.
` *`
  * **getByTitle**: The title attribute is not consistently read by screenreaders, 
  * and is not visible by default for sighted users.
  * 
  * **getByRole**: This can be used to select dialog boxes and other 
  * difficult-to-capture elements in a more semantic way
  * 
  * Test IDs
  * 
  * **getByTestId**: The user cannot see (or hear) these, so this is only 
  * recommended for cases where you can't match by text or it doesn't make 
  * sense (the text is dynamic).
  */
