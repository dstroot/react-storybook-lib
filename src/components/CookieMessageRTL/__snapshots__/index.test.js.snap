// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieMessageRTL it should disappear when the button is clicked 1`] = `null`;

exports[`CookieMessageRTL it should render 1`] = `
<div
  class="fixed-bottom bg-dark"
  data-testid="cookie-message"
>
  <table
    class="table table-borderless m-0"
  >
    <tbody>
      <tr>
        <td
          class="text-light"
        >
          We use cookies to understand the performance of our web site, enable social media features, and serve more relevant content to you. We may also use cookies on our and our partners’ behalf to help us manage advertising and assess the performance of our campaigns. For further information please see our 
          <a
            href="/cookiepolicy"
            title="Cookie Policy."
          >
            Cookie Policy
          </a>
          .
        </td>
        <td
          class="align-middle"
        >
          <button
            class="btn btn-light btn-sm"
          >
            I Agree
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
